src/services/dmr/DmrService.ts

```diff
@@ -179,17 +179,45 @@ export class DmrService {
     let msg: string = "An unspecified error has been received from the server";
     if (response.status !== 401) {
       let result: any = response.data;
-      if (result && result["failure-description"] != null) {
-        if (isNotNullOrUndefined(result["failure-description"]["domain-failure-description"])) {
-          msg = result["failure-description"]["domain-failure-description"];
-        } else {
-          msg = result["failure-description"];
+      if (isNotNullOrUndefined(result)) {
+        if (isNotNullOrUndefined(result["failure-description"])) {
+          if (isNotNullOrUndefined(result["failure-description"]["domain-failure-description"])) {
+            msg = result["failure-description"]["domain-failure-description"];
+          } else {
+            msg = result["failure-description"];
+          }
+        }
+        // see if we can find the cause of this error
+        let cause: string = this.findErrorCause(result);
+        if (isNotNullOrUndefined(cause)) {
+          msg = cause;
         }
       }
       promise.reject(msg);
     }
   }
 
+  private findErrorCause(root: any): string {
+    // traverse the object tree
+    let cause: string = undefined;
+    traverseObject(root, (key: string, value: any, trail: string []) => {
+      // NOTE "Operation step-1" contains the cause, adjust if necessary
+      let errorFound: string = this.visitTraversedErrorNode(value, "Operation step-1");
```

Comment:
No need for the variable `errorFound` or method `visitTraversedErrorNode` as only used once and just adds verbosity. Can be:

```typescript
if (isNotNullOrUndefined(value) && isNotNullOrUndefined(obj["Operation step-1"]) {
    cause = obj["Operation step-1"];
}
```

Reply:
Also, what happens if the error is on `Operation step-<x>`

Reply:
True, how would we do the search for other Operation step or if the op was not composite. I'll investigate this a bit further